src/streaming/StreamProcessor.js

```diff
@@ -184,20 +234,74 @@ function StreamProcessor(config) {
         return representationController ? representationController.isUpdating() : false;
     }
 
+    function onStreamInitialized(e) {
+        if (!e.streamInfo || streamInfo.id !== e.streamInfo.id) return;
+
+        if (!streamInitialized) {
+            streamInitialized = true;
+            if (isDynamic) {
+                timelineConverter.setTimeSyncCompleted(true);
+                setLiveEdgeSeekTarget();
+            } else {
+                const seekTarget = playbackController.getStreamStartTime(false);
+                bufferController.setSeekStartTime(seekTarget);
+                scheduleController.setCurrentRepresentation(getRepresentationInfo());
+                scheduleController.setSeekTarget(seekTarget);
+            }
+        }
+
+        scheduleController.start();
+    }
+
     function onDataUpdateCompleted(e) {
-        if (e.sender.getType() !== getType() || e.sender.getStreamId() !== streamInfo.id || !e.error || e.error.code !== Errors.SEGMENTS_UPDATE_FAILED_ERROR_CODE) return;
+        if (e.sender.getType() !== getType() || e.sender.getStreamId() !== streamInfo.id) return;
 
-        addDVRMetric();
+        if (!e.error) {
+            scheduleController.setCurrentRepresentation(adapter.convertDataToRepresentationInfo(e.currentRepresentation));
+        } else if (e.error.code !== Errors.SEGMENTS_UPDATE_FAILED_ERROR_CODE) {
```

Comment:
Compared to previous check, addDVRMetric should only be called if e.error.code === Errors.SEGMENTS_UPDATE_FAILED_ERROR_CODE

Reply:
Previous check was:
`        if (e.sender.getType() !== getType() || e.sender.getStreamId() !== getStreamInfo().id || !e.error || e.error.code !== Errors.SEGMENTS_UPDATE_FAILED_ERROR_CODE) return;
`